src/main/java/redis/clients/jedis/Pipeline.java

```diff
@@ -94,10 +99,17 @@ public boolean isInMulti() {
    * commands you execute.
    */
   public void sync() {
-    if (getPipelinedResponseLength() > 0) {
-      List<Object> unformatted = client.getMany(getPipelinedResponseLength());
-      for (Object o : unformatted) {
-        generateResponse(o);
+    try{
+      if (getPipelinedResponseLength() > 0) {
+        List<Object> unformatted = client.getMany(getPipelinedResponseLength());
+        for (Object o : unformatted) {
+          generateResponse(o);
+        }
+      }
+    }finally {
+      if(jedis != null){
```

Comment:
This means that once `pipeline.sync()`, the pipeline is no longer available, this is incompatible with the previous semantics, and it is not what I approve of.

> It is expected that the user would be careful to avoid these situations.

Agree, the user guarantees that it is correct and recommended to use `Pipeline` before Jedis is returned/closed.